include/irr/asset/CTexturePacker.h

```diff
@@ -105,31 +84,74 @@ class ITexturePacker
 
     page_tab_offset_t alloc(const ICPUImage* _img, const ICPUImage::SSubresourceRange& _subres)
     {
-        auto extent = _img->getCreationParameters().extent;
-
-        const uint32_t w = std::max(extent.width>>_subres.baseMipLevel,1u);
-        const uint32_t h = std::max(extent.height>>_subres.baseMipLevel,1u);
-        const uint32_t pgSqrSz = core::roundUpToPoT((std::max(w,h)+m_pgSzxy-1u)/m_pgSzxy);
+        const uint32_t pgSqrSz = computeSquareSz(_img, _subres);
 
         const uint32_t addr = m_pgtAddrAlctr.alloc_addr(pgSqrSz);
         return (addr==CSquareAddressAllocator<uint32_t>::invalid_address) ? 
             page_tab_offset_invalid() :
             page_tab_offset_t(m_pgtAddrAlctr.unpackAddress_x(addr), m_pgtAddrAlctr.unpackAddress_y(addr));
     }
+    virtual void free(page_tab_offset_t _addr, const ICPUImage* _img, const ICPUImage::SSubresourceRange& _subres)
```

Comment:
if you're going down the virtual function route in `ITexturePacker` then the type for `alloc` and `free` needs to be `IImage` not `ICPUImage`

Reply:
right, thx